### DISCUSSION TITLE
How can I share a magnet link to my files and read in WebTorrent with Angular?

### DISCUSSION THREAD

**Discussion by jocimarlopes** [webtorrent/webtorrent#2699] [Q&A] [answered]
How can I share a magnet link to my files?
I want to create a decentralized system, but I can only read example magnet links on WebTorrent.

the code simply does not run client.add() with an external link (it is not an illegal practice)

**Comment by SilentBot1** [comment, marked as answer]
You can seed files in one of these three ways:
- http/https web seeds
- "hybrid" peers (i.e. peers which can speak with both WebRTC peers, and regular TCP/UDP peers), some examples are:
   -  WebTorrent Desktop (via WebTorrent-Hybrid library)
   - Vuze /w WebTorrent extension
   - LibTorrent /w WebTorrent enabled.
- WebRTC peers (WebTorrent in the browser)

If you are calling `client.add` in the browser with torrents from the TCP or UDP swarms, this **will not work**, as a browser **can not** directly speak with these other peers, as they speak different communication. 

Hopefully this answers your question, but if you have any more, please feel free to ask them.

**Reply by jocimarlopes** [reply]
Thanks for you answer, you helped me a lot!
Just one thing, can I build a Android Mobile App with this code? Or just with Electron for Windows?

**Comment by jocimarlopes** [comment]
Hey bro! Another thing, I want to run an example video, but it not works.. 
look that image
It read the torrent infos, but dont found the example video (Big Buck Bunny) path in URL on localhost. 

<img width="1470" alt="Captura de Tela 2023-11-08 às 02 17 36" src="https://github.com/webtorrent/webtorrent/assets/45192419/d436d036-3066-418f-a925-58ac67d84942">

**Reply by SilentBot1** [reply]
Hey there, 

I've answered this in the other discussion you posed with a link to what I believe is the relavent issue, but TLDR, appears something is patching your fetch method, causing it to prepend `/assets/js/` to the URL, where there URL should just be `http://localhost:4200/webtorrent/...`.
